core/storage/btree.rs

```diff
@@ -1847,7 +1851,18 @@ impl Cursor for BTreeCursor {
     }
 
     fn wait_for_completion(&mut self) -> Result<()> {
-        // TODO: Wait for pager I/O to complete
+        let start = std::time::Instant::now();
```

Comment:
You need to use `IO::get_current_time()` so we can simulate this deterministically.

Reply:
Do we really want to `wait_for_completions` inside the VM? Aren't we supposed to predictably submit I/O on the outer loop of VDBE?